docs/common/HANDLERS.md

```diff
@@ -444,6 +444,96 @@ const addItemWithTimestamp = handler<
 );
 ```
 
+### Storing Pattern Instances in Cell Arrays
+
+**⚠️ CRITICAL:** When storing pattern instances (recipes) in a `Cell` array, you must use a `lift` function to perform the `.push()` operation. Calling `.push()` directly in a handler will cause a stack overflow.
+
+This pattern is commonly used to create "charm lists" or "pattern collections" (like a list of chat instances).
+
+#### ❌ WRONG - Stack Overflow
+
+```typescript
+import { handler, Cell, ID } from "commontools";
+
+type Entry = {
+  [ID]: string;
+  local_id: string;
+  charm: any;
+};
+
+// ❌ This will cause: RangeError: Maximum call stack size exceeded
+const createPattern = handler<
+  unknown,
+  { patternsList: Cell<Entry[]> }
+>(
+  (_, { patternsList }) => {
+    const pattern = MyPattern({ title: "New" });
+
+    // ❌ WRONG: Direct .push() in handler causes stack overflow
+    patternsList.push({
+      [ID]: randomId,
+      local_id: randomId,
+      charm: pattern
+    });
+  }
+);
+```
+
+#### ✅ CORRECT - Use lift()
+
+```typescript
+import { handler, Cell, ID, lift, toSchema, cell, OpaqueRef } from "commontools";
+
+type Entry = {
+  [ID]: string;
+  local_id: string;
+  charm: any;
+};
+
+// ✅ CORRECT: Wrap the push operation in a lift function
+const storePattern = lift(
+  toSchema<{
+    charm: any;
+    patternsList: Cell<Entry[]>;
+    isInitialized: Cell<boolean>;
+  }>(),
+  undefined,
+  ({ charm, patternsList, isInitialized }) => {
+    if (!isInitialized.get()) {
+      // ✅ CORRECT: .push() happens inside lift, not handler
+      patternsList.push({
+        [ID]: Math.random().toString(36).substring(2, 10),
+        local_id: Math.random().toString(36).substring(2, 10),
+        charm
+      });
+      isInitialized.set(true);
+    }
+  }
+);
+
+// Handler returns the lift call instead of pushing directly
+const createPattern = handler<
+  unknown,
+  { patternsList: Cell<Entry[]> }
+>(
+  (_, { patternsList }) => {
+    const isInitialized = cell(false);
+    const pattern = MyPattern({ title: "New" });
+
+    // ✅ CORRECT: Return the lift function call
+    return storePattern({
+      charm: pattern,
+      patternsList: patternsList as unknown as OpaqueRef<Entry[]>,
```

Comment:
I don't think this casting is necessary anymore either, and if it is, the big type refactor that is pending ought to fix this.